src/ngraph/builder/quantization.cpp

```diff
@@ -0,0 +1,162 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "ngraph/builder/quantization.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/dequantize.hpp"
+#include "ngraph/op/quantized_avg_pool.hpp"
+#include "ngraph/op/quantized_conv.hpp"
+#include "ngraph/op/quantized_conv_bias.hpp"
+#include "ngraph/op/quantized_conv_relu.hpp"
+#include "ngraph/op/quantized_max_pool.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+namespace ngraph
+{
+    namespace builder
+    {
+        std::shared_ptr<Node> ScaledQuantizedAvgPool(const std::shared_ptr<Node>& arg,
+                                                     const Shape& window_shape,
+                                                     const Strides& window_movement_strides,
+                                                     const Shape& padding_below,
+                                                     const Shape& padding_above,
+                                                     bool include_padding_in_avg_computation,
+                                                     const std::shared_ptr<Node> min,
+                                                     const std::shared_ptr<Node> max)
+        {
+            // TODO: make a builder to calculate
+            auto scale = op::Constant::create(element::f32, Shape{}, {1});
+
+            return make_shared<op::QuantizedAvgPool>(arg,
+                                                     window_shape,
+                                                     window_movement_strides,
+                                                     padding_below,
+                                                     padding_above,
+                                                     include_padding_in_avg_computation,
+                                                     scale);
```

Comment:
We dont need scale for avg pool since we dont requantize.

Reply:
Yes, correct. Adam and me had this discussion. Will make the changes.